tools/run-integration-test.sh

```diff
@@ -0,0 +1,186 @@
+    mv ~/.timesketchrc ~/.timesketchrc.$DATE
+    mv ~/.timesketch.token ~/.timesketch.token.$DATE
+fi
+
+echo "Generating Timesketch config..."
+TS_SECRET=$(kubectl get secret --namespace default $RELEASE-timesketch-secret -o jsonpath="{.data.timesketch-user}" | base64 -d)
+if ! [ -f ~/.timesketchrc ] && ! [ -f ~/.timesketch.token ]; then
+    expect <<END_EXPECT
+    spawn timesketch config
+    expect "What is the value for <host_uri>"
+    send "http://127.0.0.1:5000\r"
+    expect "What is the value for <auth_mode>"
+    send "userpass\r"
+    expect "What is the value for <username>"
+    send "timesketch\r"
+    expect "Password for user timesketch"
+    send "$TS_SECRET\r"
+    expect eof
+END_EXPECT
+fi
+
+# Back up existing Turbinia config else script will attempt to connect to wrong Turbinia instance
+if  [ -f ~/.turbinia_api_config.json ]
+then
+  echo "Backing up existing Turbinia config to ~/.turbinia_api_config.json.$DATE"
+  mv ~/.turbinia_api_config.json ~/.turbinia_api_config.json.$DATE
+fi
+
+# Replace Turbinia config with test config
+echo "Writing turbinia config to $HOME/.turbinia_api_config.json..."
+cat > $HOME/.turbinia_api_config.json <<EOL
+{
+	"default": {
+		"description": "Turbinia client test config",
+		"API_SERVER_ADDRESS": "http://127.0.0.1",
+		"API_SERVER_PORT": 8000,
+		"API_AUTHENTICATION_ENABLED": false,
+		"CLIENT_SECRETS_FILENAME": ".client_secrets.json",
+		"CREDENTIALS_FILENAME": ".credentials_default.json"
+	}
+}
+EOL
+
+# Run dfTimewolf recipe
+echo "Running dfTimewolf recipe: dftimewolf gcp_turbinia_ts $GCP_PROJECT $GCP_ZONE --disk_names $DISK --incident_id test213 --timesketch_username timesketch --timesketch_password TS_SECRET"
+export DFTIMEWOLF_NO_CURSES=1
+dftimewolf gcp_turbinia_ts $GCP_PROJECT $GCP_ZONE --disk_names $DISK --incident_id test213 --timesketch_username timesketch --timesketch_password $TS_SECRET
+echo "dfTimewolf recipe succeeded!" 
+
+# Turbinia integration test
+echo "Starting integration test for Turbinia..."
+
+# Grab all Turbinia request IDs
+REQUEST_SUMMARY=$(turbinia-client status summary -j)
+REQUEST_IDS=$(echo $REQUEST_SUMMARY | jq -r '.[] | .[].request_id')
+for req in $REQUEST_IDS
+do
+	# Grab all PlasoParserTask or PlasoHasherTask where successful = false
+  echo "Checking the status of Turbinia request: $req"
+  status=$(turbinia-client status request $req -j)
+	plaso_status=$(echo $status | jq '[.tasks[]] | map({name: .name, id: .id, successful: .successful, worker_name: .worker_name}) | map(select(.name == "PlasoParserTask" or .name == "PlasoHasherTask")) | map(select(.successful==false))')
+	length=$(echo $plaso_status | jq '. | length')
+	if [[ $length > 0 ]]
+	then
+	  echo "A failed Plaso Task for Turbinia request $req has been detected."
+	  echo "Listing failed Tasks..."
+    # Grab the Task ID
+    tasks=$(echo $plaso_status | jq -r '.[] | .id')
+	  FAILED=1
+	  for t in $tasks
```

Comment:
I'm not a big fan of one letter variable names. I opt to call this `task` instead

Reply:
Fair enough, I updated any one character variable (including this) to be more explicit